src/daft-functions/src/distance/dot.rs

```diff
@@ -0,0 +1,129 @@
+use common_error::{DaftError, DaftResult};
+use daft_core::{
+    datatypes::{Field, Float64Array},
+    schema::Schema,
+    DataType, IntoSeries, Series,
+};
+use daft_dsl::{
+    functions::{ScalarFunction, ScalarUDF},
+    ExprRef,
+};
+use serde::{Deserialize, Serialize};
+use simsimd::SpatialSimilarity;
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq, Hash)]
+pub struct DotDistanceFunction {}
+
+macro_rules! compute_dot_distance {
+    ($source:expr, $query:expr, $type:ident) => {{
+        let query = &$query.fixed_size_list()?.flat_child;
+        let query = query.$type()?.as_slice();
+        $source
+            .into_iter()
+            .map(|list_opt| {
+                let list = list_opt?;
+                let list = list.$type().unwrap().as_slice();
+                let cosine = $type::dot(&list, &query);
+                cosine
+            })
+            .collect::<Vec<_>>()
+    }};
+}
+
+#[typetag::serde]
+impl ScalarUDF for DotDistanceFunction {
```

Comment:
we can probably just have 1 DistanceFunction but then parameterize similarly to our trig functions.

https://github.com/Eventual-Inc/Daft/blob/c24635e74f4c69f83bcc6958b8031eeef9043295/src/daft-dsl/src/functions/numeric/mod.rs#L68

Reply:
the `distance/dot` was supposed to be part of a separate PR. I removed this file